src/st/collect_heart_vessel.h

```diff
@@ -0,0 +1,14 @@
+void CollectHeartVessel(void) {
+    if (g_PlayableCharacter != PLAYER_ALUCARD) {
+        g_api.PlaySfx(NA_SE_PL_COLLECT_HEART);
+        g_Status.hearts += HEART_VESSEL_RICHTER;
+
+        if (g_Status.heartsMax < g_Status.hearts) {
+            g_Status.hearts = g_Status.heartsMax;
+        }
```

Comment:
Tiny nit: I think this can be simplified to `CLAMP_MAX(g_Status.hearts, g_Status.heartsMax);`

https://decomp.me/scratch/0D98F